.scripts/git-branch.ps1

```diff
@@ -0,0 +1,74 @@
+[CmdletBinding()]
+param
+(
+    [Parameter(Mandatory = $true)]
+    [string]$Name,
+    
+    [Parameter(Mandatory = $true)]
+    [string]$Email
+)
+
+function TagExists($TagName) {
+    
+    git tag | ForEach-Object {
+
+        if ($_ -eq $TagName) {
+            return $true
+        }
+    }
+
+    return $false
+}
+
+$date = Get-Date -UFormat "%Y-%m-%d"
+$branch = "release_$date"
+
+$sources = $env:BUILD_SOURCESDIRECTORY
+
+Write-Output "Set current directory to '$sources'"
+Set-Location $sources
+
+git config user.email $Email
+git config user.name $Name
+
+$message = "Bump package versions for release $date"
+
+Write-Output "Create new branch '$($branch)'"
+
+git checkout -b $branch
+
+$lastCommitId = git rev-list --tags --max-count=1
+$lastTag = git describe $lastCommitId  # By default git describe only shows annotated tags.
+
+if ($LASTEXITCODE -ne 0) {
+
+    git checkout master
+    git branch -d
+
+    throw "Failed to find last tagged commit ($LASTEXITCODE)"
+}
+
+Write-Output "Print commits from $lastTag..$branch"
+git log $lastTag..$branch --name-status
+
+git add .
+git commit -m  $message
+
+if ($(TagExists -TagName $date) -eq $false) {
+
+    Write-Output "Create new tag '$($date)'"
+    git tag -a $date -m "SDK release on $date"
+}
+else {
+    
+    Write-Output "Tag '$($date)' already exists"
```

Comment:
>Write-Output "Tag '$($date)' already exists" [](start = 4, length = 44)

at that point is' probably best to fail and exit.

Reply:
ok but...why do we need to fail here?  I could be missing something but in testing this case works as expected

---
In reply to: [267187667](https://github.com/Azure/azure-iot-sdk-node/pull/510#discussion_r267187667) [](ancestors = 267187667)

Reply:
added comment to explain this, pls correct me if I'm mistaken

---
In reply to: [267483066](https://github.com/Azure/azure-iot-sdk-node/pull/510#discussion_r267483066) [](ancestors = 267483066,267187667)